woocommerce/woocommerce-ios
# Refactor and update how we wait for a created site to be ready

Parts of #9733 


## Why


As `StoreCreationCoordinator` already got too huge with many feature flags, it's getting harder to test some of the logic (the coordinator shouldn't be responsible for making API requests anyways). In this PR, the logic that waits for the site to be ready was extracted from `StoreCreationCoordinator` to its own class `StoreCreationStatusChecker` since there are also some updates.

## How

Previously, the sequence of network requests is like:
- `sites/:siteId/plugins` up to 10 times until Jetpack plugin is active
- `sites/:siteId` to fetch the site, and throw an error `newSiteIsNotFullySynced` once when it's missing some properties (it's a known issue on the backend pe5sF9-1vz-p2#comment-2184), and an event is tracked
- `sites/:siteId` to fetch the site again and return success

Now with the latest conclusion in pe5sF9-1vz-p2#comment-2180, the sequence would be:
- `sites/:siteId/plugins` up to 10 times until Jetpack **and Woo** plugins are both active
- `sites/:siteId` to fetch the site and return success
  - at the call site `StoreCreationCoordinator`, it checks if it's missing some properties like before and track the same event

All the other behaviors should stay the same, including local notifications and progress UI.


## Testing instructions


- Log in with a WPCOM account if needed
- Go to the Menu tab, and tap on the store picker
- At the end of the store list, tap `Add a store`
- Continue the steps to create a free trial site --> the progress UI should be incremented like before. after about 30-60 seconds, the flow should succeed and navigate to the newly created store without a Jetpack upsell banner, with these two events logged `login_woocommerce_site_created` (with a `waiting_time` property) and `site_creation_properties_out_of_sync`

---
- [x] @jaclync tests local notifications work as before

---
